src/sentry/static/sentry/app/views/organizationEventsV2/tags.jsx

```diff
@@ -1,18 +1,90 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import {isEqual} from 'lodash';
 
 import SentryTypes from 'app/sentryTypes';
 import TagDistributionMeter from 'app/components/tagDistributionMeter';
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import {fetchTags, getEventTagSearchUrl} from './utils';
 
-export default class Tags extends React.Component {
+class Tags extends React.Component {
   static propTypes = {
+    api: PropTypes.object.isRequired,
+    organization: SentryTypes.Organization.isRequired,
     view: SentryTypes.EventView.isRequired,
+    selection: SentryTypes.GlobalSelection.isRequired,
+    location: PropTypes.object.isRequired,
+  };
+
+  state = {
+    isLoading: true,
+    tags: {},
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.view.id !== prevProps.view.id ||
+      !isEqual(this.props.selection, prevProps.selection)
+    ) {
+      this.fetchData();
+    }
+  }
+
+  fetchData = async () => {
+    const {api, organization, view} = this.props;
+
+    try {
+      const tags = await fetchTags(api, organization.slug, view.tags);
+      this.setState({tags, isLoading: false});
+    } catch (err) {
+      this.setState({tags: {}, isLoading: false});
```

Comment:
Should we show a component level error in this case?

Reply:
Good point, will follow up with error states